source/server/config/http/router.h

```diff
@@ -18,6 +21,15 @@ class RouterFilterConfig : public NamedHttpFilterConfigFactory {
   HttpFilterFactoryCb createFilterFactory(const Json::Object& json_config,
                                           const std::string& stat_prefix,
                                           FactoryContext& context) override;
+
+  HttpFilterFactoryCb createFilterFactoryFromProto(const Protobuf::Message& config,
+                                                   const std::string& stat_prefix,
+                                                   FactoryContext& context) override;
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
```

Comment:
I think this PR has the same issue as @rshriram PR in that this path actually isn't tested. I think he can fix in his other PR since we need to do it for all filters. @htuch not sure if you have any thoughts on how best to test generically.